tests/system_tests/test_api_endpoints.py

```diff
@@ -0,0 +1,190 @@
+import time
+from typing import Any
+
+import pytest
+import requests
+from fastapi import status
+from pydantic import TypeAdapter
+
+from blueapi.client.rest import BlueapiRestClient, BlueskyRemoteControlError
+from blueapi.service.interface import get_devices, get_plans
+from blueapi.service.model import (
+    DeviceResponse,
+    EnvironmentResponse,
+    PlanResponse,
+    TasksListResponse,
+    WorkerTask,
+)
+from blueapi.worker.event import TaskStatusEnum, WorkerState
+from blueapi.worker.task import Task
+from blueapi.worker.task_worker import TrackableTask
+
+_SIMPLE_TASK = Task(name="sleep", params={"time": 0.0})
+_LONG_TASK = Task(name="sleep", params={"time": 1.0})
+
+rest = BlueapiRestClient()
+
+
+def get_response(
+    url: str, BaseModel: Any, status_code: int = status.HTTP_200_OK
+) -> Any:
+    get_response = requests.get(url)
+    model = TypeAdapter(BaseModel).validate_python(get_response.json())
+    assert get_response.status_code == status_code
+    return model
+
+
+def test_get_plans():
+    assert rest.get_plans() == PlanResponse(plans=get_plans())
+
+
+def test_get_plans_by_name():
+    for plan in get_plans():
+        assert rest.get_plan(plan.name) == plan
+
+
+def test_get_non_existant_plan():
+    with pytest.raises(KeyError) as exception:
+        rest.get_plan("Not exists")
+    assert exception.value.args[0] == ("{'detail': 'Item not found'}")
+
+
+def test_get_devices():
+    assert rest.get_devices() == DeviceResponse(devices=get_devices())
+
+
+def test_get_device_by_name():
+    for device in get_devices():
+        assert rest.get_device(device.name) == device
+
+
+def test_get_non_existant_device():
+    with pytest.raises(KeyError) as exception:
+        assert rest.get_device("Not exists")
+    assert exception.value.args[0] == ("{'detail': 'Item not found'}")
+
+
+def test_create_task_and_delete_task_by_id():
+    create_task = rest.create_task(_SIMPLE_TASK)
+    rest.clear_task(create_task.task_id)
+
+
+def test_create_task_validation_error():
+    with pytest.raises(KeyError) as exception:
+        rest.create_task(Task(name="Not-exists", params={"Not-exists": 0.0}))
+    assert exception.value.args[0] == ("{'detail': 'Item not found'}")
+
+
+def test_get_all_tasks():
+    created_tasks = []
+    for task in [_SIMPLE_TASK, _LONG_TASK]:
+        created_task = rest.create_task(task)
+        created_tasks.append(created_task)
+
+    task_list = get_response(rest._url("/tasks"), TasksListResponse)
```

Comment:
I'm unsure why the client object doesn't provide what you want here

Reply:
I have added code for this here https://github.com/DiamondLightSource/blueapi/pull/618/commits/b5bb82524a2a8b3e1852254f0ef6f2d52d85dcb3 

I think we should change 

```
get_task(task_id)         to   get_task_by_id(task_id) -> assert task_id is not ""

and get_all_tasks()  
```

Reply:
It was added by this https://github.com/DiamondLightSource/blueapi/pull/508